src/include/cache.h

```diff
@@ -183,7 +183,8 @@ struct __wt_cache {
      */
     int32_t evict_hs_score;
 
-    uint32_t hs_fileid; /* History store table file ID */
+    uint64_t hs_counter; /* History store counter */
```

Comment:
I'm concerned that using a global counter here will be a bottleneck. Can we use a per-btree counter instead?

I think it must be OK to have the counter reset, since it will be reset after a restart regardless. We could use the write generation - that's a counter we already have.

Reply:
af97158

As for write generation, I don't think we can do that. Every page reconciliation is going to have its own unique write gen. So in the case that multiple transactions commit at the same timestamp, they will have the same write gen and therefore the same history store key when reconciling.